ADR/0016-integration-service-promotion-logic.md

```diff
@@ -0,0 +1,93 @@
+# 16. Promotion logic in the Integration Service
+
+Date Documented: 2023-02-08
+
+## Status
+
+In Review
+
+## Context
+
+Before the merge of HACBS & AppStudio, the AppStudio build-service created the
+ApplicationSnapshot and promoted the content to the user's lowest environment as soon as
+the build was completed and once the environment was built.
+The integration-service fulfilled the same role for HACBS in addition to running the 
+IntegrationTestScenario tests first before promoting.
+
+Here, we are documenting the decision to consolidate and move the logic at one place to avoid duplicacy
+on the actions.
+
+Note: This functionality has now been completely dropped from the build-service side after the merger.
+
+### What is promotion logic ?
+
+DevOps workflows often automate deployments of applications across different environments to ensure 
+that the workloads are properly tested before being further promoted to an environment with a higher 
+service level agreement. 
+The promotion path can be represented with a directed acyclic graph from the environment with the 
+lowest SLA to the one with the highest, for example development -> staging -> production. 
+In Stonesoup, this promotion logic would be represented by a set of components (container images) defined by 
+an immutable Snapshot being deployed to the relevant environment.
+
+Once the Snapshot is tested and verified successfully, its contents will then be deployed to the user's 
+lowest environments.
+
+The “lowest environments” are the Environments with no explicit parentEnvironment set.
+
+## Decision
+
+Consolidate the promotion logic for both HACBS & AppStudio and move it to the integration-service after 
+the merger of Stonesoup.
+
+![](../diagrams/ADR-0016/promotion-logic.jpg)
+
+Below are the three steps to follow in the integration-service to validate and promote the content 
+of the successful Snapshot:
+
+* Ensures that the ContainerImage in the Global Candidate List(GCL) is being updated when the Snapshot 
+  passes all the integration tests.
```

Comment:
Is the integration service actually ensuring that the Global Candidate List is updated? Is it performing a check to make sure that the tests passed and that the list matches the Snapshot or are these just "guarantees" because the Snapshot status has been changed to `successful` (or whatever the actual state is) instead of `invalid`?

If there is no actual work being done to ensure this, then the language may be imprecise -- either in the bullet or the lead-in sentence above it.

Reply:
ack, adjusted the wording of the statement